src/material.js

```diff
@@ -319,4 +319,31 @@ export class Material extends HasMetadataNamedDefaultableInMemoryEntity {
 
         return new this.constructor(config);
     }
+
+    /**
+     * @summary a series of checks for the material and returns an array of results.
+     * @returns {*[]} - Array of checks results
+     */
+    runChecks() {
+        const checks = [];
+        const basis = this.Basis;
+
+        const overlappingGroups = basis.getOverlappingAtoms();
+        overlappingGroups.forEach((group) => {
+            group.forEach((atom) => {
+                const otherAtoms = group
+                    .filter((other) => other.id !== atom.id)
+                    .map((other) => `${other.element} (Line ${other.id + 1})`)
+                    .join(", ");
+
+                checks.push({
+                    type: "warning",
+                    message: `${atom.element} (Line ${atom.id + 1}) overlaps with ${otherAtoms}`,
```

Comment:
I'm thinking turning this into more generic message from an enum, to use like `messages.atomsOverlap` and define it all in a dedicated property.